components/content_settings/renderer/brave_content_settings_agent_impl.cc

```diff
@@ -135,6 +144,58 @@ void BraveContentSettingsAgentImpl::DidNotAllowScript() {
   ContentSettingsAgentImpl::DidNotAllowScript();
 }
 
+bool BraveContentSettingsAgentImpl::UseEphemeralStorageSync(
+    StorageType storage_type) {
+  if (!base::FeatureList::IsEnabled(net::features::kBraveEphemeralStorage))
+    return false;
+
+  blink::WebLocalFrame* frame = render_frame()->GetWebFrame();
+
+  if (!frame || IsFrameWithOpaqueOrigin(frame))
+    return false;
+
+  return // block 3p
+         !ContentSettingsAgentImpl::AllowStorageAccessSync(storage_type) &&
+         // allow 1p
+         AllowStorageAccessForMainFrameSync(storage_type);
+}
+
+bool BraveContentSettingsAgentImpl::AllowStorageAccessSync(
```

Comment:
still need to update the non-sync versions of these methods just for consistency in the results

Reply:
also in particular this needs a test to make sure it only applies to local/session storage